modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/index/hash/io/HashIndexTreeIo.java

```diff
@@ -166,23 +227,51 @@ default HashIndexRow getRow(DataPageReader dataPageReader, int partitionId, long
             throws IgniteInternalCheckedException {
         int off = offset(idx);
 
-        int hash = getInt(pageAddr, off + INDEX_COLUMNS_HASH_OFFSET);
+        int hash = getInt(pageAddr, off);
+        off += Integer.BYTES;
+
+        short indexColumnsSize = getShort(pageAddr, off);
+        off += Short.BYTES;
+
+        ByteBuffer indexColumnsBuffer;
+
+        long link;
 
-        long link = readPartitionless(partitionId, pageAddr, off + INDEX_COLUMNS_LINK_OFFSET);
+        if (indexColumnsSize < 0) {
+            indexColumnsBuffer = ByteBuffer.wrap(getBytes(pageAddr, off, -indexColumnsSize));
+            off += -indexColumnsSize;
```

Comment:
Please use proper value here as well, RowId should always be at the fixed position

Reply:
fix it